servicesCore/requestCore_service.py

```diff
@@ -1,35 +1,219 @@
+                dataResult['data'] = await self._ipfs.get_file(dataResult.data)
+            else:
+                dataResult['data'] = None
+
+            return dataResult
+        except Exception as e:
+            raise e
+
+    async def getRequestByTransactionHash(self, _hash:str):
+        """
+        Get a request and method called by the hash of a transaction
+        :param hash: hash of the ethereum transaction
+        """
+        try:
+            errors = []
+            warnings = []
+            transaction = await self._web3Single.getTransaction(_hash)
+            if not transaction:
+                raise ValueError('transaction not found')
+
+            ccyContract = transaction.to
+
+            ccyContractservice = await servicesContracts.getServiceFromAddress(ccyContract)
+            # get information from the currency contract
+            if not ccyContractservice:
+                raise ValueError('Contract is not supported by request')
+
+            method = ccyContractservice.decodeInputData(transaction.input)
+            if not method.name:
+                raise ValueError('transaction data not parsable')
+
+            request = None
+
+            txReceipt = await self._web3Single.getTransactionReceipt(_hash)
+            # if already mined
+            if txReceipt:
+                if txReceipt.status != '0x1' and txReceipt.status != 1:
+                    errors.append('transaction has failed')
+                elif transaction.method and transaction.method.pararmeters and transaction.method.parameters._requestId:
+                    # simple action
+                    request = await self.getRequest(transaction.method.parameters._requestId)
+                elif transaction txReceipt.logs and txReceipt.logs[0] and self._web3Single.areSameAddressNoChecksum(txReceipt.logs[0].address, self._addressRequestCore)
+                    # maybe a creation
+                    event = self._web3Single.decodeTransactionLog(self._abiRequestCore, 'Created', txReceipt.logs[0])
+                    if event:
+                        request = self.getRequest(event.requestId)
+            else:
+                # if not mined
+                methodGenerated = ccyContractservice.generateWeb3Method(transaction.method.name, self._web3Single.resultToArray(transaction.method.parameters))
+                options = {
+                    'from': transaction.from,
+                    'gas': transaction.gas,
+                    'value': transaction.value
+                }
+
+                try:
+                    test = await self._web3Single.callMethod(methodGenerated, options)
+                except Exception as e:
+                    warnings.append('transaction may failed: ' + e)
+
+                if transaction.gasPrice < config.ethereum.gasPriceMinimumCriticalInWei:
+                    warnings.append('transaction gasPrice is low')
+
+            errors = None if not errors else errors
```

Comment:
You can do `errors = errors or None` and if `errors == []` it will be `None`. Same for warnings below.